starlette_admin/contrib/beanie/view.py

```diff
@@ -0,0 +1,343 @@
+                elif isinstance(field, HasOne):
+                    if action == RequestAction.EDIT:
+                        obj_serialized[field.name] = (
+                            await foreign_model.get_serialized_pk_value(request, value)
+                        )
+                    else:
+                        obj_serialized[field.name] = await foreign_model.serialize(
+                            value.ref, request, action, include_relationships=False
+                        )
+                else:
+                    if action == RequestAction.EDIT:
+                        obj_serialized[field.name] = [
+                            (await foreign_model.get_serialized_pk_value(request, obj))
+                            for obj in value
+                        ]
+                    else:
+                        obj_serialized[field.name] = [
+                            await foreign_model.serialize(
+                                v.ref, request, action, include_relationships=False
+                            )
+                            for v in value
+                        ]
+            elif not isinstance(field, RelationField):
+                value = await field.parse_obj(request, obj)
+                obj_serialized[field.name] = await self.serialize_field_value(
+                    value, field, action, request
+                )
+        if include_select2:
+            obj_meta["select2"] = {
+                "selection": await self.select2_selection(obj, request),
+                "result": await self.select2_result(obj, request),
+            }
+        obj_meta["repr"] = await self.repr(obj, request)
+
+        # Make sure the primary key is always available
+        pk_attr = not_none(self.pk_attr)
+        if pk_attr not in obj_serialized:
+            pk_value = await self.get_serialized_pk_value(request, obj)
+            obj_serialized[pk_attr] = pk_value
+
+        pk = await self.get_pk_value(request, obj)
+        route_name = request.app.state.ROUTE_NAME
+        obj_meta["detailUrl"] = str(
+            request.url_for(route_name + ":detail", identity=self.identity, pk=pk)
+        )
+        obj_serialized["_meta"] = obj_meta
+        return obj_serialized
+
+    async def create(self, request: Request, data: dict) -> Document:
+        data = {
+            k: v for k, v in data.items() if k not in self.exclude_fields_from_create
+        }
+        try:
+            doc = self.document(**data)
+        except ValidationError as ve:
+            raise pydantic_error_to_form_validation_errors(ve) from ve
+        return await doc.create()
+
+    async def edit(
+        self, request: Request, pk: PydanticObjectId, data: dict
+    ) -> Document:
+        doc: Document | None = await self.document.get(pk)
+        assert doc is not None, "Document not found"
+        data = {k: v for k, v in data.items() if k not in self.exclude_fields_from_edit}
```

Comment:
you should use `self.get_fields_list()` instead

Reply:
Updated in 7083157